src/download_target.cpp

```diff
@@ -34,11 +35,53 @@ namespace powerloader
         if (m_is_zchunk)
         {
             m_p_zck = std::make_unique<zck_target>();
-            m_p_zck->zck_cache_file = filename;
+            m_p_zck->zck_cache_file = m_destination_path;
         }
 #endif
     }
 
+    std::shared_ptr<DownloadTarget> DownloadTarget::from_url(Context& ctx,
+                                                             const std::string& target_url,
+                                                             const fs::path& destination_path,
+                                                             const fs::path& destination_dir)
+    {
+        if (contains(target_url, "://"))
+        {
+            // even when we get a regular URL like `http://test.com/download.tar.gz`
+            // we want to create a "mirror" for `http://test.com` to make sure we correctly
+            // retry and wait on mirror failures
+            URLHandler uh{ target_url };
+            const std::string url = uh.url();
+            const std::string host = uh.host();
+            const std::string path = uh.path();
+            const std::string mirror_url = url.substr(0, url.size() - path.size());
```

Comment:
There is already a function for computing this in [URLHandler](https://github.com/mamba-org/powerloader/blob/280389379fcd37c9ac0f42ed4a3c2530f4705d87/include/powerloader/url.hpp#L74)